### PR TITLE
Create IoT Firmware Defense SDK

### PR DESCRIPTION
**Pull request by TheDonLuke**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/fist/Azure.ResourceManager.IoTFirmwareDefense/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Microsoft Azure IoTFirmwareDefense management client library for .NET
+
+**[Describe the service briefly first.]**
```
Comment (ArcturusZhang) [inline sdk/fist/Azure.ResourceManager.IoTFirmwareDefense/README.md:3]: Could we include some introduction of your RP here?

Comment (eatkimchee) [reply sdk/fist/Azure.ResourceManager.IoTFirmwareDefense/README.md]: Hi @ArcturusZhang the readme has been updated, and we added tests as well. Can you review again please? thanks!

**Comment by m-nash** [comment]
There is no ci run for mgmt iot we definitely need to fix that before this goes out since there is no apiview to review here.

**Review by m-nash — COMMENTED** [review]
The capitalism for `IoT` should be `Iot` when used in a compound word.

https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-namespace-naming

**Comment by m-nash** [comment]
Left some comments in apiview https://apiview.dev/Assemblies/Review/3a9374b65a5b41a58cf6c16f8435ed97/4594a9acd08046e7a3153c0742e1cd5e

**Comment by eatkimchee** [comment]
@m-nash   we have addressed things from the apiview review and left some responses for you. Can you check it again with the latest build?

**Comment by m-nash** [comment]
> @m-nash we have addressed things from the apiview review and left some responses for you. Can you check it again with the latest build?

Left a few replies and I have one follow up item on the `bool` vs extensible enum I will get back to you on.  Nothing in my opinion is a blocker for beta with the couple additional prefix type names that I commented on.  Everything else can be covered in GA timeframe.

**Comment by eatkimchee** [comment]
@m-nash added the prefixes you asked for

**Comment by eatkimchee** [comment]
@ArcturusZhang  Hi can you please approve this PR and merge it if we met all the requirements? thanks!

**Review by ArcturusZhang — COMMENTED** [review]
Generally looks good to me, but I did not see the api file in your PR, could you please run the script `{Root}/eng/Export-API.ps1 iot` to generate it and push it to your branch?

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**